doc/guide/javascript.rst

```diff
@@ -0,0 +1,234 @@
+          return $(block).prop('xblock_children');
+        },
+        childMap: function(block, childName) {
+          var children = this.children(block);
+          for (var i = 0; i < children.length; i++) {
+            var child = children[i];
+            if (child.name == childName) {
+              return child
+            }
+          }
+        }
+      }
+    };
+    return {
+      getRuntime: getRuntime,
+      versions: versions
+    };
+  }());
+
+
+The Runtime Handler
+*********************
+
+The JavaScript runtime initializes the XBlock each time it is loaded by
+a user and returns the handler so the XBlock can communicate with the server.
+
+From the example above, the following part of the runtime generates and returns
+the handler to the XBlock:
+
+.. code-block:: JavaScript
+
+    . . .
+
+    var versions = {
+      1: {
+        handlerUrl: function(block, handlerName, suffix, query) {
+        suffix = typeof suffix !== 'undefined' ? suffix : '';
+        query = typeof query !== 'undefined' ? query : '';
+        var usage = $(block).data('usage');
+        var url_selector = $(block).data('url_selector');
+        if (url_selector !== undefined) {
+            baseUrl = window[url_selector];
+        }
+        else {baseUrl = handlerBaseUrl;}
+
+          // studentId and handlerBaseUrl are both defined in block.html
+          return (baseUrl + usage +
+                           "/" + handlerName +
+                           "/" + suffix +
+                   "?student=" + studentId +
+                           "&" + query);
+
+    . . . 
+
+The runtime handler code is called by the XBlock's JavaScript code to get the XBlock URL. 
+
+For example, in the example `Thumbs XBlock`_ in the `XBlock SDK`_, the
+`thumbs.js`_ file gets the handler from the XBlock runtime:
+
+.. code-block:: JavaScript
+
+    var handlerUrl = runtime.handlerUrl(element, 'vote');
+
+
+XBlock Children
+*********************
+
+The JavaScript runtime also returns the list of child XBlocks to the XBlock.
+
+From the example above, the following part of the runtime generates and returns
+the list of children to the XBlock:
+
+.. code-block:: JavaScript
+
+    . . .
+
+    children: function(block) {
+          return $(block).prop('xblock_children');
+        },
+    . . . 
+
+NEED EXAMPLE OF XBLOCK USING THIS
+
+WHEN XBLOCK USES THIS VS. MAP
+
+XBlock Child Map
+*********************
+
+The JavaScript runtime also returns the a map of child XBlocks to the running
+XBlock.
+
+From the example above, the following part of the runtime generates and returns
+the list of children to the XBlock:
+
+.. code-block:: JavaScript
+
+    . . .
+
+    childMap: function(block, childName) {
+      var children = this.children(block);
+      for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        if (child.name == childName) {
+          return child
+        }
+      }
+    }
+    . . . 
+
+NEED EXAMPLE OF XBLOCK USING THIS
```

Comment:
An XBlock uses this when it needs to perform different actions on different children. The only example of this that I'm aware of is the `problem_demo` block in the XBlock SDK.